File: app/src/main/java/com/cn29/aac/ui/shoppingkart/vm/ShoppingKartActivityViewModelFactory.java
package com.cn29.aac.ui.shoppingkart.vm;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

/**
 * Created by Charles Ng on 25/10/2017.
 */

public class ShoppingKartActivityViewModelFactory extends ViewModelProvider.NewInstanceFactory {

  @Inject
  public ShoppingKartActivityViewModelFactory() {
    super();
  }

  @NonNull
  @Override
  public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
    return (T) new ShoppingKartActivityViewModel();
  }
}

File: app/src/main/java/com/cn29/aac/ui/shoppingkart/vm/ShoppingKartActivityViewModelFactory.kt
package com.cn29.aac.ui.shoppingkart.vm

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import javax.inject.Inject

/**
 * Created by Charles Ng on 25/10/2017.
 */
class ShoppingKartActivityViewModelFactory @Inject constructor() :
        NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return ShoppingKartActivityViewModel() as T
    }
}